File: 단기스터디뿌셔뿌셔/java/prog_나머지가 1이 되는 수 찾기.java
class Solution {
    public int solution(int n) {
        int idx = 2;
        while(!(n % idx == 1)) {
            ++idx;
        }
        return idx;
    }
}

import java.util.*;
import java.util.stream.IntStream;

// stream은 함수비용이 적은 코드에서는 성능이 좋지 않음
class Solution {
    public int solution(int n) {
        return IntStream.range(2, n).filter(idx -> n % idx == 1).findFirst().orElse(0);
    }
}

File: C++/programmers/level1/prog_나머지가 1이되는 수 찾기.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(int n) {
    int answer = n;
    for(int idx = 2; idx < n; idx++){
        if(n % idx == 1){
            answer = min(answer, idx);
        }
    }
    return answer;
}